### ISSUE TITLE
Nuxt.js [Vue warn]: Unknown custom element: <affix>

### ISSUE THREAD

**Issue by NetBzz** [eddiemf/vue-affix#15]
Sadly can't get the plugin to work in Nuxt.js

Navigation.vue component
```
    <template>
      <affix class="top-nav" relative-element-selector="#content">
        <nav id="navigation" role="navigation" aria-label="main navigation">
          <a href="#test">Test</a>
        </nav>
      </affix>
    </template
```

nuxt.config.js
```
module.exports = {
  plugins: [
    { src: '~/plugins/vue-affix', srr: false }
  ]
}
```

vue-affix.js
```
import Vue from 'vue'
var Affix = require('vue-affix');

Vue.use(Affix);
```

**Comment by appinteractive** [comment]
You have a typo in ssr (ssr). 
And you probably want to wrap the component with <no-ssr></no-ssr>

**Comment by Lempert** [comment]
Just tried to install the plugin and had a similar problem without nuxt.
Then I changed 'var Affix = require('vue-affix');'  to 'import':
import Affix from 'vue-affix';
It sounds silly but it solved my problem.

**Comment by eddiemf** [comment]
Have you had any success on getting this to work? I'm not sure if it's because of your typo or if it's really a bug.

Thanks for the feedback!

**Comment by appinteractive** [comment]
@eddiemf for me at least it worked like a charm. @Lempert, @NetBzz I also used import instead of require. With require you sometimes may have to add .default to the end.

**Comment by scoutrul** [comment]
Hello! I'm using this plugin with Nuxt too and had same problem. 
You should add info about Nuxt installation to the doc.
This is my way: 
1) add Affix plugin to nuxt config 
```
 import Affix from 'vue-affix';
 Vue.use(Affix)
```
2) use tag 'affix'

**Comment by yvess** [comment]
with nuxt 2.12 I need to do this:

### nuxt.config.js

```
...
  plugins: [
    { src: '~/plugins/vue-affix', mode: 'client' }
  ],
...
```

### plugins/vue-affix.js

```
import Vue from 'vue'
import Affix from 'vue-affix'

if (process.client) {
  Vue.use(Affix)
}
```

### in template 

```
...
      <client-only>
        <affix
          relative-element-selector="#content"
          style="width: 400px"
        >
        ...
       </affix>
     </client-only>
...
```

**Comment by appinteractive** [comment]
The `<client-only/>` part might be really bad depending on the content rendered inside as it will not get indexed by search machines properly. The package should account for this and just output the slot content when `window` is not available.

**Comment by yvess** [comment]
> The `<client-only/>` part might be really bad depending on the content rendered inside as it will not get indexed by search machines properly. The package should account for this and just output the slot content when `window` is not available.

Yes it would be better if the package would properly support SSR

**Comment by CavalcanteLeo** [comment]
No plans to support SSR?
